### PR TITLE
v1.12 Backports 2023-12-20

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
* [x] #29809 (@marseel)
 * [x] #29880 (@julianwiedmann)

⚠️ Conflict observed, see backporter's note for more details
 * [x] #22384 (@shaardie)
 * [x] #29616 (@learnitall)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 22384 29809 29880 29616
```

### CODE REVIEW

**Review by julianwiedmann — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
/test-backport-1.12

**Comment by YutaroHayakawa** [comment]
k8s-1.24-kernel-net-next: https://github.com/cilium/cilium/issues/18285
k8s-1.19-kernel-4.9: https://github.com/cilium/cilium/issues/27118
k8s-1.16-kernel-4.9: https://github.com/cilium/cilium/issues/27118

**Comment by YutaroHayakawa** [comment]
/test-1.24-net-next

**Comment by YutaroHayakawa** [comment]
/test-1.19-4.9

**Comment by YutaroHayakawa** [comment]
/test-1.16-4.9

**Comment by YutaroHayakawa** [comment]
/test-1.20-4.9

File: Documentation/spelling_wordlist.txt [hunk 1]
```diff
@@ -2,6 +2,10 @@ AArch
 AKS
 Alemayhu
 Alibaba
+Anthos
+Automount
+BIGTCP
+BPF
```
Comment (sayboras) [inline Documentation/spelling_wordlist.txt]: we probably only need autoamount in this commit.

**Review by sayboras — APPROVED** [review]
Small comment for https://github.com/cilium/cilium/pull/22384, but it's not a blocker.

**Review by marseel — APPROVED** [review]

**Review by learnitall — CHANGES_REQUESTED** [review]
Changes LGTM, but I'm not sure I understand the backporter note that you left. It says:

```
[ backporter's note: backporter's note: Conflict due to the
  do_netdev_encrypt_encap doesn't exist on v1.12 branch. Removed it. ]
```

But the change on line 757 in `bpf/bpf_host.c` is inside the function `do_netdev_encrypt_encap`. Maybe I'm misunderstanding something?

**Comment by YutaroHayakawa** [comment]
@learnitall Ah, sorry. I was misunderstanding the conflict. In the main branch. `do_netdev_encrypt_encap` does exist in `v1.12`, but it is just wrapped with `do_netdev_encrypt`. In the main branch, `do_netdev_encrypt` has gone, and `do_netdev_enctypt is directly called here.

https://github.com/cilium/cilium/blob/0e9a54251888b40fed5bda94e66d08294666e6fa/bpf/bpf_host.c#L1081C21-L1087

I'll fix the note for more clarity.

File: Documentation/spelling_wordlist.txt (hunk 1 above)
Comment (YutaroHayakawa) [reply Documentation/spelling_wordlist.txt]: Thanks! Fixed in the latest push.

**Review by learnitall — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**